### ISSUE TITLE
"Trying to remove nonexistant room build callback" freeze/crash

### ISSUE THREAD

**Issue by NikNaks** [CorsixTH/CorsixTH#858]
##### What steps will reproduce the problem?
1. Run Autosave7.sav
2. Wait a few seconds
3. The game will freeze/crash
##### What version of CorsixTH are you using (e.g. "0.30", "4c66e39985")?

0.5 RC1
##### What operating system / compile settings are you using?

Windows 10
##### What level was this on (e.g. "Demo level", "Full game level 12")?

Level 10 (Eggsenham)
##### Please provide any additional information below.

Uploads of gamelog and autosaves here: https://db.tt/AgUDUY76
I've included Autosave6 as well, in case this helps shed light on what changed between the two saves to cause the crash.
##### gamelog.txt output

```
Created game with savegame version 105.
Warning: Trying to remove nonexistant room build callback (function: 0000000F2E6AFC80) from humanoid (table: 0000000F2ED67EE0).
Warning: Trying to remove nonexistant room remove callback (function: 0000000F2A485F80) from humanoid (table: 0000000F2ED67EE0).
[repeated mulitple times]
Warning: Trying to remove nonexistant room build callback (function: 0000000F2E159FE0) from humanoid (table: 0000000F23930A00).
Warning: Trying to remove nonexistant room remove callback (function: 0000000F2A343B40) from humanoid (table: 0000000F23930A00).
Error in timer handler: 
C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:662: Can only finish current action
stack traceback:
    [C]: in function 'assert'
    C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:662: in method 'finishAction'
    C:\Program Files\CorsixTH\Lua\rooms\general_diag.lua:75: in field 'after_use'
    ...Files\CorsixTH\Lua\humanoid_actions\multi_use_object.lua:264: in local 'timer_function'
    C:\Program Files\CorsixTH\Lua\entity.lua:218: in method 'tick'
    C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
    C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
    (...tail calls...)
    C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>
```

**Comment by Alberth289346** [comment]
Cannot reproduce this in current trunk (a54ddfa, October 2015). Did you try in 0.50 release?

**Comment by mugmuggy** [comment]
Fixed with #1567
